### ISSUE TITLE
SockProtocol: No support for redundant protocol values from different families

### ISSUE THREAD

**Issue by fpagliughi** [nix-rust/nix#1903]
I wanted to send in a PR to add the CANbus socket protocols to `SockProtocol`, to include `CanRaw` (1) and `CanBcm` (2).

Like this:
```
    /// The Controller Area Network raw socket protocol
    /// ([ref](https://docs.kernel.org/networking/can.html#how-to-use-socketcan))
    #[cfg(any(target_os = "android", target_os = "linux"))]
    #[cfg_attr(docsrs, doc(cfg(all())))]
    CanRaw = libc::CAN_RAW,
    /// The Controller Area Network broadcast manager protocol
    /// ([ref](https://docs.kernel.org/networking/can.html#how-to-use-socketcan))
    #[cfg(any(target_os = "android", target_os = "linux"))]
    #[cfg_attr(docsrs, doc(cfg(all())))]
    CanBcm = libc::CAN_BCM,
```

The problem is that CAN_BCM = 2, which is already used by the discriminant, `NetlinkUserSock = libc::NETLINK_USERSOCK`.

Should `SockProtocol` be converted into an enum of enum's for each protocol family?

**Comment by asomers** [comment]
Annoying.  But no, it shouldn't be an enum of enums.  That would require either making separate `socket` functions for each AddressFamily, or using tagged enums, which aren't zero-cost.  Instead, we should either use `libc_bitflags!`, or if this is the only violator we can add a `const CanBcm = SockProtocol::NetlinkUserSock` alias.

**Comment by fpagliughi** [comment]
Yeah, what I was imagining was a tagged outer enum, which, true, isn't zero cost, but seems to model the issue properly: Each protocol _family_ has it's own enumeration of different protocols.

For that matter, though, we could probably infer both the family and protocol from the tagged outer enum! So one family/protocol parameter instead of two. That would even out the cost, but introduce some new issues.

It's just pretty funny that there wasn't an overlap with all the values you already implemented.

But, anyway, I'm not looking to make this perfect here. I'm looking to solve an immediate couple of problems. The quick-fix to make `CanBcm` an external constant works well enough for me.

**Comment by fpagliughi** [comment]
OK, PR #1912 is up for this, as well as #1914 which provides a `CanAddr` to bind the CAN socket.
